### PR TITLE
feat: add users to mls conversation (FS-851)

### PR DESCRIPTION
**Pull request by aweiss-dev**
BREAKING:

- needed to change the addUsers() params
- split up Proteus and MLS functions to reduce complexity

### CODE REVIEW

File: packages/api-client/src/conversation/ConversationAPI/AddUsers/AddUsers.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+}
+
+/**
+ * Add qualified members to an existing MLS conversation.
+ * @param groupId The conversation ID to add the users to
+ * @param users List of users to add to a conversation
+ */
+export async function addMLSUsers(this: ConversationAPI, groupId: string, users: string[]) {
+  console.info('addMLSUsers', groupId, users);
```
Comment (aweiss-dev) [inline packages/api-client/src/conversation/ConversationAPI/AddUsers/AddUsers.ts:103]: Note to myself. Delete this func. As we found out it is not necessary

File: packages/api-client/src/conversation/ConversationAPI/ConversationAPI.ts [hunk 2]
```diff
@@ -959,12 +959,11 @@ export class ConversationAPI {
...
   /**
-   * Add users to an existing conversation.
+   * Add qualified members to an existing conversation.
    * @param conversationId The conversation ID to add the users to
-   * @param userIds List of user IDs to add to a conversation
-   * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/conversations/addMembers
+   * @param users List of users to add to a conversation
    */
-  private async postMembersV0(conversationId: string, userIds: string[]): Promise<ConversationMemberJoinEvent> {
+  public async postMembersV0(conversationId: string, userIds: string[]) {
```
Comment (tlebon) [inline packages/api-client/src/conversation/ConversationAPI/ConversationAPI.ts:966]: i may prefer `postMembersProteus` as a name here

File: packages/api-client/src/conversation/ConversationAPI/ConversationAPI.ts [hunk 3]
```diff
@@ -959,12 +959,11 @@ export class ConversationAPI {
   }
 
   /**
-   * Add users to an existing conversation.
+   * Add qualified members to an existing conversation.
```
Comment (tlebon) [inline packages/api-client/src/conversation/ConversationAPI/ConversationAPI.ts:962]: i dont believe qualified members is the correct description as this is just a userId array and not a qualified id array. In general the naming in this file is hard to differentiate for me.

File: packages/api-client/src/conversation/ConversationAPI/ConversationAPI.ts [hunk 4]
```diff
@@ -959,12 +959,11 @@ export class ConversationAPI {
...
    * @param conversationId The conversation ID to add the users to
-   * @param userIds List of user IDs to add to a conversation
-   * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/conversations/addMembers
+   * @param users List of users to add to a conversation
    */
-  private async postMembersV0(conversationId: string, userIds: string[]): Promise<ConversationMemberJoinEvent> {
+  public async postMembersV0(conversationId: string, userIds: string[]) {
     const config: AxiosRequestConfig = {
```
Comment (aweiss-dev) [inline packages/api-client/src/conversation/ConversationAPI/ConversationAPI.ts:967]: Needs to be private

File: packages/bot-api/src/MessageHandler.ts [hunk 5]
```diff
@@ -49,10 +49,9 @@ export abstract class MessageHandler {
 
   async addUser(conversationId: QualifiedId, userId: string) {
     if (this.account?.service) {
-      await this.account.service.conversation.addUsers({
+      await this.account.service.conversation.addUsersToProteusConversation({
         conversationId,
-        userIds: [{domain: '', id: userId}],
-        protocol: ConversationProtocol.PROTEUS,
+        qualifiedUserIds: [{domain: '', id: userId}],
```
Comment (tlebon) [inline packages/bot-api/src/MessageHandler.ts:54]: question, why are we stripping off the qualifiedId that we pass in here? seems the domain will be relevant if it exists

Comment (tlebon) [reply packages/bot-api/src/MessageHandler.ts:54]: ive noticed this in a few other spots recently, wondering if it could be causing any inconsistencies

**Review by tlebon — COMMENTED** [review]
Lets please wait a little to merge PRs this big in the future!

**Pull request merged**